### ISSUE TITLE
Support for cookie-based authentication for servers behind a reverse proxy

### ISSUE THREAD

**Issue by pauldiloreto** [Suwayomi/Tachidesk-Sorayomi#291]
**Is your feature request related to a problem? Please describe.**
Unless I'm mistaken, the only way to connect to your Tachidesk server when outside of the local network is to use basic auth, or simply use a VPN to securely access the local network. But many of us home lab users already have several services running behind a reverse proxy with an authentication layer (i.e. Authelia, Authentik, etc.) in order to securely expose them to the internet. I believe this is a common use case that should be supported.

Until this is supported, it will not be possible to reach Tachidesk servers that are secured behind more modern security protocols. It would be great to unblock the ability to use features like LDAP, Open ID Connect, MFA, etc...  I'd love to be able to turn this on for my Tachidesk instance!

**Describe the solution you'd like**
In order to support cookie-based authentication, there needs to be a way to "Open in Browser" (sort of like what already exists in the Server settings page), but in this case it's in order to log in to the redirected authentication portal. After successfully logging in, the browser typically sets a cookie (used for authenticating the rest of the session) & redirects to originally requested URL.

tldr; open browser to log in & persist cookie plz?

**Describe alternatives you've considered**
Just use basic auth? :(
I could use a VPN to just be on my local network securely, but that's a hassle since Sorayomi is what I'd use on my mobile devices. Not impossible, but just inconvenient when cookie-based authentication is already set up and pretty standard.

**Additional context**
Nope, that should sum it up!

**Comment by tomerh2001** [comment]
> **Is your feature request related to a problem? Please describe.**
> 
> Unless I'm mistaken, the only way to connect to your Tachidesk server when outside of the local network is to use basic auth, or simply use a VPN to securely access the local network. But many of us home lab users already have several services running behind a reverse proxy with an authentication layer (i.e. Authelia, Authentik, etc.) in order to securely expose them to the internet. I believe this is a common use case that should be supported.
> 
> Until this is supported, it will not be possible to reach Tachidesk servers that are secured behind more modern security practices. It would be great to unblock the ability to use features like LDAP, Open ID Connect, MFA, etc...  I'd love to be able to turn this on for my Tachidesk instance!
> 
> **Describe the solution you'd like**
> 
> In order to support cookie-based authentication, there needs to be a way to "Open in Browser" (sort of like what already exists in the Server settings page), but in this case it's in order to log in to the redirected authentication portal. After successfully logging in, the browser typically sets a cookie (used for authenticating the rest of the session) & redirects to originally requested URL.
> 
> tldr; open browser to log in & persist cookie plz?
> 
> **Describe alternatives you've considered**
> 
> Just use basic auth? :(
> I could use a VPN to just be on my local network securely, but that's a hassle since Sorayomi is what I'd use on my mobile devices. Not impossible, but just inconvenient when cookie-based authentication is already set up and pretty standard.
> 
> **Additional context**
> Nope, that should sum it up!
> 

Any updates on this?

**Comment by DattatreyaReddy** [comment]
sorry ill look in to this

**Comment by gKits** [comment]
+1 Since Suwayomi-Server doesn't natively support external authentication services (yet) it would be at least really helpful to allow the Sorayomi app to authenticate itself via the already existing JWT authentication used by the `ui_login` mode.
